lib/database/classes.ts

```diff
@@ -63,37 +79,141 @@ const updateClass = async (
+      "FROM (((event_information e INNER JOIN classes cl ON e.id = cl.event_information_id) " +
+      "INNER JOIN commitments ON commitments.event_information_id = e.id) " +
+      " INNER JOIN users u ON commitments.user_id = u.id) WHERE role = 'Teacher' AND cl.event_information_id = $1",
     values: [id],
   };
-
   const res = await client.query(query);
+
   if (res.rows.length == 0) {
     return null;
   }
 
-  let oneClass: Class;
+  let classesWithUserInformation: ClassWithUserInformation[];
   try {
-    oneClass = await decode(ClassSchema, res.rows[0]);
+    classesWithUserInformation = await decode(ClassWithUserInformationArraySchema, res.rows);
   } catch {
     throw Error("Fields returned incorrectly in database");
   }
+  //console.log(classesWithUserInformation);
+  const mapOfClasses = new Map();
+  const individualClasses = [];
+  for (let i = 0; i < classesWithUserInformation.length; i++) {
+    if (!mapOfClasses.has(classesWithUserInformation[i].eventInformationId)) {
+      const currClass = {
+        name: classesWithUserInformation[i].name,
+        eventInformationId: classesWithUserInformation[i].eventInformationId,
+        minLevel: classesWithUserInformation[i].minLevel,
+        maxLevel: classesWithUserInformation[i].maxLevel,
+        rrstring: classesWithUserInformation[i].rrstring,
+        startTime: classesWithUserInformation[i].startTime,
+        endTime: classesWithUserInformation[i].endTime,
+        language: classesWithUserInformation[i].language,
+      };
+      individualClasses.push(currClass);
+      const curr = {
+        userId: classesWithUserInformation[i].userId,
+        firstName: classesWithUserInformation[i].firstName,
+        lastName: classesWithUserInformation[i].lastName,
+      };
+      const teachersArr = [curr];
+      mapOfClasses.set(classesWithUserInformation[i].eventInformationId, teachersArr);
+    } else {
+      const curr = {
+        userId: classesWithUserInformation[i].userId,
+        firstName: classesWithUserInformation[i].firstName,
+        lastName: classesWithUserInformation[i].lastName,
+      };
+      const teachersArr = mapOfClasses.get(classesWithUserInformation[i].eventInformationId);
+      teachersArr.push(curr);
 
-  return oneClass;
+      mapOfClasses.set(classesWithUserInformation[i].eventInformationId, teachersArr);
+    }
+  }
+  for (const singleClass of individualClasses) {
+    const currClass: Class = {
+      name: singleClass.name,
+      eventInformationId: singleClass.eventInformationId,
+      minLevel: singleClass.minLevel,
+      maxLevel: singleClass.maxLevel,
+      rrstring: singleClass.rrstring,
+      startTime: singleClass.startTime,
+      endTime: singleClass.endTime,
+      language: singleClass.language,
+      teachers: mapOfClasses.get(singleClass.eventInformationId),
+    };
+    return currClass;
+  }
+  return null;
```

Comment:
i dont think we need to do any of this for a singular class. correct? We are just returning a single class, so having to do all these iterations seems redundant